ddcommon/src/container_id.rs

```diff
@@ -107,12 +120,47 @@ pub fn get_container_id() -> Option<&'static str> {
     // cache container id in a static to avoid recomputing it at each call
 
     lazy_static! {
-        static ref CONTAINER_ID: Option<String> =
-            extract_container_id(get_cgroup_path().as_path()).ok();
+        static ref CONTAINER_ID: Option<String> = {
+            let old_id = extract_container_id(get_cgroup_path().as_path()).ok();
+            // fallback to the new Id retrieval mechanism
+            old_id.or_else(|| CGroupSurrogateId::read().ok().map(|i| i.as_header_value()) )
+        };
     }
     CONTAINER_ID.as_deref()
 }
 
+pub struct CGroupSurrogateId {
+    pub device: u64,
+    pub inode: u64,
+}
+
+impl CGroupSurrogateId {
+    fn read_from_path<P: AsRef<Path>>(path: P) -> Result<Self, ContainerIdError> {
+        let path = path.as_ref();
+        let meta = std::fs::metadata(path).map_err(|e| ContainerIdError::ErrorAccessingFile {
+            path: path.to_path_buf(),
+            source: e,
+        })?;
+
+        let inode = meta.ino();
+        let device = meta.dev();
+
+        Ok(CGroupSurrogateId {
+            device: device,
+            inode: inode,
+        })
+    }
+
+    pub fn read() -> Result<Self, ContainerIdError> {
+        Self::read_from_path(DEFAULT_SYS_CGROUP_PATH)
```

Comment:
I think we need to verify if `sys/fs/cgroup` is always the cgroup node of the container. In our tests it was always the case but IIRC the spec says that`/proc/<pid>/cgroup` contains the exact path: https://man7.org/linux/man-pages/man7/cgroups.7.html